modules/cudaimgproc/src/mssegmentation.cpp

```diff
@@ -388,7 +388,20 @@ void cv::cuda::meanShiftSegmentation(InputArray _src, OutputArray _dst, int sp,
             dstcol[3] = 255;
         }
     }
-    dst.copyTo(_dst);
+
+    if (_dst.kind() == _InputArray::CUDA_GPU_MAT)
+    {
+        GpuMat dstGpuMat = getOutputMat(_dst, src.size(), src.type(), stream);
+        if (stream)
+            dstGpuMat.upload(dst, stream);
+        else
+            dstGpuMat.upload(dst);
+    }
+    else {
+        _dst.create(src.size(), src.type());
+        Mat dstMat = _dst.getMat();
+        dst.copyTo(dstMat);
```

Comment:
> 3 lines

Did you try just `dst.copyTo(_dst);` call?

Reply:
No, but it works perfectly, I wrongly assumed that this worked the same for `Mat` and `GpuMat`.

Reply:
I believe it makes sense to improve `GpuMat` support in `.copyTo()` implementation (in a separate PR).

Reply:
OK I can take a look at that once this has been merged.